File: kanji_app/lib/models/tentativa.dart
class Tentativa {
  final String id;
  final String usuarioId;
  final String? kanjiId;
  final String? palavraId;
  final bool acertou;
  final DateTime data;

  Tentativa({
    required this.id,
    required this.usuarioId,
    this.kanjiId,
    this.palavraId,
    required this.acertou,
    required this.data,
  });

  factory Tentativa.fromJson(Map<String, dynamic> json) {
    return Tentativa(
      id: json['_id'],
      usuarioId: json['usuario_id'],
      kanjiId: json['kanji_id'],
      palavraId: json['palavra_id'],
      acertou: json['acertou'],
      data: DateTime.parse(json['data']),
    );
  }
}

File: backend/models/Tentativa.js
const mongoose = require('mongoose');

const tentativaSchema = new mongoose.Schema({
  data: { type: Date, default: Date.now },
  acertou: Boolean,
  usuario_id: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  kanji_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Kanji' }, // sorteado
  palavra_id: { type: mongoose.Schema.Types.ObjectId, ref: 'Palavra' } // opcional
});

module.exports = mongoose.model('Tentativa', tentativaSchema);
